benchmark/poseidon/poseidon_benchmark_runner.h

```diff
@@ -0,0 +1,79 @@
+#ifndef BENCHMARK_POSEIDON_POSEIDON_BENCHMARK_RUNNER_H_
+#define BENCHMARK_POSEIDON_POSEIDON_BENCHMARK_RUNNER_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+// clang-format off
+#include "benchmark/poseidon/poseidon_benchmark_config.h"
+#include "benchmark/poseidon/simple_poseidon_benchmark_reporter.h"
+// clang-format on
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/base/logging.h"
+#include "tachyon/base/time/time.h"
+#include "tachyon/cc/math/finite_fields/prime_field_traits_forward.h"
+#include "tachyon/crypto/hashes/sponge/poseidon/poseidon.h"
+
+namespace tachyon {
+
+template <typename Field>
+class PoseidonBenchmarkRunner {
+ public:
+  using CPrimeField = typename cc::math::PrimeFieldTraits<Field>::CPrimeField;
+
+  typedef CPrimeField* (*PoseidonExternalFn)(const CPrimeField* pre_images,
+                                             size_t absorbing_num,
+                                             size_t squeezing_num,
+                                             uint64_t* duration);
+
+  explicit PoseidonBenchmarkRunner(SimplePoseidonBenchmarkReporter* reporter,
+                                   PoseidonBenchmarkConfig* config)
+      : reporter_(reporter), config_(config) {
+    pre_images_ = base::CreateVector(config->absorbing_num(),
+                                     []() { return Field::Random(); });
```

Comment:
I just wanted to ask you, this is also fine, though. Did you do this intentionally instead of `pre_images_(base::CreateVector(...))`

Reply:
Yeah, I thought this has better readability!